sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamJoinTransforms.java

```diff
@@ -154,17 +151,14 @@ private static Row combineTwoRowsIntoOne(Row leftRow,
    */
   private static Row combineTwoRowsIntoOneHelper(Row leftRow, Row rightRow) {
     // build the type
-    List<String> names = new ArrayList<>(leftRow.getFieldCount() + rightRow.getFieldCount());
-    names.addAll(leftRow.getRowType().getFieldNames());
-    names.addAll(rightRow.getRowType().getFieldNames());
-
-    List<Coder> types = new ArrayList<>(leftRow.getFieldCount() + rightRow.getFieldCount());
-    types.addAll(leftRow.getRowType().getRowCoder().getCoders());
-    types.addAll(rightRow.getRowType().getRowCoder().getCoders());
-    RowType type = RowType.fromNamesAndCoders(names, types);
+    List<Schema.Field> fields = new ArrayList<>(
+        leftRow.getFieldCount() + rightRow.getFieldCount());
+    fields.addAll(leftRow.getSchema().getFields());
+    fields.addAll(rightRow.getSchema().getFields());
+    Schema type = Schema.of(fields);
```

Comment:
<!--thread_id:cc_178372816_t; commit:79c95678e593da730ba0472b77304ec1f916245e; resolved:0-->
<!--section:context-quote-->
> **akedin** wrote:
> Couple of questions related to this piece:
>  1. Should schema internally support operations like this? E.g. `schema1.join(schema2)`?
>  2. It probably makes sense to cache these schemas eventually, I would leave a TODO;
>  3. Even if schema doesn't support complex operations, probably makes sense to have a complex builder: `Schema.builder().addFields(schema1.getFields()).addFields(schema2.getFields()).build()`;

<!--section:body-->
Yes, at the very least we should have helper functions. Could be on schema, could be on SchemaBuilder, or could be a separate SchemaHelpers class.